### DISCUSSION TITLE
BadRequestError: request aborted while calling /health endpoint

### DISCUSSION THREAD

**Discussion by kumarimanjari** [chimurai/http-proxy-middleware#790] [Q&A]
### Checks

- [X] I understand project setup issues should be asked on [StackOverflow](https://stackoverflow.com/questions/tagged/http-proxy-middleware) or in [GitHub Discussions](https://github.com/chimurai/http-proxy-middleware/discussions).
- [X] I updated to latest `http-proxy-middleware`.

### Describe the bug (be clear and concise)

We have an express server where /health endpoint is present (GET method). 

There is an external system which tries to call /health endpoint so that it can check the status and process further.
But whenever other system tries to call it, we see below error in our log

BadRequestError: request aborted
[HPM] Error occurred while proxying request <source_url> to <target_url>
[ECONNRESET] (https://nodejs.org/api/errors.html#errors_common_system_errors)

### Step-by-step reproduction instructions

```shell
1. calling the /health endpoint from a different system
```

### Expected behavior (be clear and concise)

We should be able to get 200 status HTTP code, whenever we /health endpoint is called.

### How is http-proxy-middleware used in your project?

```shell
We have an express server created as below: 
app.use('*', <middleware>, createProxyMiddleware({target, changeOrigin, onProxyReq: <function>, selfHandleResponse: true, onProxyRes: <function> })

Version -> http-proxy-middleware - ^2.0.0, I know this is not the latest version, but I tried that as well, it didn't solved the issue.
```

### What http-proxy-middleware configuration are you using?

```typescript
createProxyMiddleware({
    target,
    changeOrigin,
    onProxyReq: ****,
    selfHandleResponse: true,
    onProxyRes: ****,
  })
```

### What OS/version and node/version are you seeing the problem?

```shell
Node Version -> v14.16.0-alpine3.13
```

**Comment by chimurai** [comment]
Sounds like a project issue.

Without minimal reproduction there is little I can do.

**Reply by kumarimanjari** [reply]
It is always getting produced in my Environment logs. How can I show you?

**Reply by chimurai** [reply]
You can create a tiny project in which this problem can be reproduced.

**Reply by kumarimanjari** [reply]
I also thought this before but to reproduce this error in local environment is nearly impossible as of now

**Reply by chimurai** [reply]
You could try `v3@beta` (`npm install http-proxy-middleware@beta`), which has some improved logging and debugging.

Maybe you'll see a little more information to troubleshoot.

https://github.com/chimurai/http-proxy-middleware#logger-object:

```js
createProxyMiddleware({
  logger: console,
});
```

https://github.com/chimurai/http-proxy-middleware#debugging:

```sh
DEBUG=http-proxy-middleware* node server.js
```

**Reply by kumarimanjari** [reply]
Sure!! I can try this. I will give an update once I tried above points

**Comment by kumarimanjari** [comment]
@chimurai  - Could you please help me here. I just want health endpoint not to be proxied, below is my code
```
app.use(
  '*',
  createProxyMiddleware(['/api/**', '!/api/health'], {
    target,
    changeOrigin,
    onProxyReq: rewrite,
    selfHandleResponse: true,
    onProxyRes: <somefunction>(cache),
    logLevel: 'debug',
  })
);
```

**Reply by chimurai** [reply]
Did you try this suggestion? And share the output? https://github.com/chimurai/http-proxy-middleware/discussions/790#discussioncomment-2981010

There is little to debug or troubleshoot with the minimal information you are sharing.

**Reply by kumarimanjari** [reply]
I tried updating http-proxy-middleware to v3.0.0 beta one, even after updating I don't see any option of logger or even pathFilter, so then I followed this link - https://www.npmjs.com/package/http-proxy-middleware#context-matching
But not able to solve. 
I have a /api/health endpoint in my express server, and of a team calls this /api/health end point to get response as 200, but they always get 504 or 404, and in logs I see BadRequest request aborted. So my question is below
1. Am I thinking correct that /api/health end point should not be proxied. If yes, then how that can be achieved

**Reply by kumarimanjari** [reply]
@chimurai - Could you please give me some hints how to move forward?

**Reply by chimurai** [reply]
v3 is not yet released as stable. The documentation you see on npmjs is still v2.

For v3 documentation you can use: https://github.com/chimurai/http-proxy-middleware

There are a lot of factors which can affect and lead to the issue you are facing; ie. client software/configuration, server software/configuration, your network and network configuration, etc...

Here you'll find notable changes and instructions to use the improved logging and debugging: https://github.com/chimurai/http-proxy-middleware/discussions/768
